pipelines/curate_bridge.py

```
@@ -579,31 +502,51 @@ def _quarantine(
     )
 
 
+def _resolve_options(
+    function_name: str,
+    supplied: dict[str, Any],
+    defaults: dict[str, Any],
+) -> dict[str, Any]:
+    unexpected = sorted(set(supplied).difference(defaults))
+    if unexpected:
+        raise TypeError(f"{function_name}() got an unexpected keyword argument {unexpected[0]!r}")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)